Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -906,9 +907,22 @@ void GraphicsContextSkia::setLineCap(LineCap lineCap)
     m_skiaState.m_stroke.cap = toSkiaCap(lineCap);
 }
 
+static bool isValidDashArray(const DashArray& dashArray)
+{
+    // See 'dom-context-2d-setlinedash': if the array contains not finite or negative values, return
+    DashArray::value_type total = 0;
+    for (const auto& dash : dashArray) {
+        if (dash < 0 || !std::isfinite(dash))
+            return false;
+        total += dash;
+    }
+    // Nothing to be done for all-zero or empty arrays
+    return total > 0;
+}
+
 void GraphicsContextSkia::setLineDash(const DashArray& dashArray, float dashOffset)
 {
-    if (dashArray.isEmpty()) {
+    if (isValidDashArray(dashArray)) {
```

Comment:
shouldn't it be `!isValidDashArray(dashArray)` ?

Reply:
Indeed it should! I started writing this as a `isDashArrayAllZeros` and as I updated it to cover all the criteria from the spec I forgot to change the check :sweat_smile: